### PR TITLE
Fix handing/reacting to changes in data entry tabs

### PR DESCRIPTION
**Pull request by siemiatj**
Related to #10125

### CODE REVIEW

File: frontend/src/actions/WindowActions.js [hunk 1]
```diff
@@ -229,6 +232,17 @@ export function unselectTab(scope) {
   };
 }
 
+/**
+ * @method setUpdatedTabLayout
+ * @summary Action creator to update tab's layout data in the store
+ */
+function setUpdatedTabLayout(tabId, layoutData) {
```
Comment (petrican) [inline frontend/src/actions/WindowActions.js:239]: I would put the params in an object ({ tabId, layoutData })

File: frontend/src/actions/WindowActions.js [hunk 2]
```diff
@@ -431,6 +443,24 @@ export function fetchTab({ tabId, windowId, docId, query }) {
   };
 }
 
+/*
+ * @method updateTabLayout
+ * @summary Action creator for fetching and updating single tab's layout
+ */
```
Comment (petrican) [inline frontend/src/actions/WindowActions.js:449]: Here too doc param and pass them as object params

File: frontend/src/containers/MasterWindow.js [hunk 3]
```diff
@@ -177,20 +181,20 @@ class MasterWindowContainer extends PureComponent {
       tabId: activeTabId,
     });
 
-    const tabLayout =
-      master.layout.tabs &&
-      master.layout.tabs.filter((tab) => tab.tabId === activeTabId)[0];
-
-    const orderBy = tabLayout.orderBy;
+    const tabLayout = getTab(master.layout, activeTabId);
+    const orderBy = tabLayout ? tabLayout.orderBy : null;
```
Comment (petrican) [inline frontend/src/containers/MasterWindow.js:185]: what happens if orderBy is undefined ?

**Review by petrican — CHANGES_REQUESTED** [review]
Some small changes requested. Other than that it is fine. Thanks!

File: frontend/src/containers/MasterWindow.js (hunk 3 above)
Comment (siemiatj) [reply frontend/src/containers/MasterWindow.js:185]: Nothing, we won't set the sorting order when requesting data.

**Pull request merged**